src/main/java/com/redhat/exhort/integration/providers/ProvidersBodyPlusResponseCodeStrategy.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.redhat.exhort.integration.providers;
+
+import java.util.Map;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.processor.aggregate.AbstractListAggregationStrategy;
+
+public class ProvidersBodyPlusResponseCodeStrategy
+    extends AbstractListAggregationStrategy<Map<Integer, String>> {
+  @Override
+  public Map<Integer, String> getValue(Exchange exchange) {
+    return Map.of(
+        exchange.getMessage().getHeader(Exchange.HTTP_RESPONSE_CODE, Integer.class),
```

Comment:
Maybe you can create a POJO to hold the providerName, the status and message to show it in the check details.
```
 "data": {
                "check.kind": "READINESS",
                "osv-nvd": "500 - Unable to validate osv-nvd Token: onguard: nodename nor servname provided, or not known",
                "snyk": "200 - OK"
            },
```

Reply:
@ruromero i can , but then i'll have to add  @RegisterForReflection annotation for this new POJO...
in the meantime, i'll add all the missing details using a map ( like i did in previous implementation), and then maybe we'll discuss it so i'll incorporate such a POJO with required fields as members...
WDYT?

Reply:
Using my approach, now it looks like this:

```json
{
    "status": "UP",
    "checks": [
        {
            "name": "camel-routes",
            "status": "UP"
        },
        {
            "name": "External Providers Readiness Check",
            "status": "UP",
            "data": {
                "oss-index": "{responseBody=Invalid token provided. Unauthorized, isEnabled=false, StatusCode=401}",
                "osv-nvd": "{responseBody=Exception occurred during execution on the exchange: Exchange[A36E02E84B42677-0000000000000002], isEnabled=true, StatusCode=500}",
                "snyk": "{responseBody=Token validated successfully, isEnabled=true, StatusCode=200}",
                "check.kind": "READINESS"
            }
        },
        {
            "name": "context",
            "status": "UP",
            "data": {
                "context.phase": "5",
                "context.name": "camel-4",
                "context.version": "4.1.0",
                "context.status": "Started",
                "check.kind": "ALL"
            }
        },
        {
            "name": "camel-consumers",
            "status": "UP"
        }
    ]
}
```

Reply:
if you're insisting, i can use a POJO and an annotation.
I Think using Exchange.HTTP_RESPONSE_TEXT as raw http response body is enough, using what you did in ProviderResponseHandler#processResponseError is an overkill :)

Reply:
I pointed out this method as reference. Sometimes the response_text is empty and you must retrieve the details from the exception and sometimes the exception is a wrapper of an httpclientException.